packages/gatsby-remark-images-contentful/src/index.js

```diff
@@ -0,0 +1,186 @@
+    wrapperStyle: ``,
+    backgroundColor: `white`,
+    linkImagesToOriginal: true,
+    showCaptions: false,
+    pathPrefix,
+  }
+
+  const options = _.defaults(pluginOptions, defaults)
+
+  // This will only work for markdown syntax image tags
+  const markdownImageNodes = select(markdownAST, `image`)
+
+  // This will also allow the use of html image tags
+  const rawHtmlNodes = select(markdownAST, `html`)
+
+  const generateImagesAndUpdateNode = async function(node, resolve) {
+    // Ingonre if it is not contentful image
+    if (node.url.indexOf(`images.ctfassets.net`) === -1) {
+      return resolve()
+    }
+
+    const metaReader = sharp()
+
+    const response = await axios({
+      method: `GET`,
+      url: `https:${node.url}`, // for some reason there is a './' prefix
+      responseType: `stream`,
+    })
+
+    response.data.pipe(metaReader)
+
+    const metadata = await metaReader.metadata()
+
+    response.data.destroy()
+
+    const responsiveSizesResult = buildResponsiveSizes({
+      metadata,
+      imageUrl: `https:${node.url}`,
+      options,
+    })
+    // Calculate the paddingBottom %
+    const ratio = `${(1 / responsiveSizesResult.aspectRatio) * 100}%`
+
+    const fallbackSrc = `https${node.url}`
+    const srcSet = responsiveSizesResult.srcSet
+    const presentationWidth = responsiveSizesResult.presentationWidth
+
+    // Generate default alt tag
+    const srcSplit = node.url.split(`/`)
+    const fileName = srcSplit[srcSplit.length - 1]
+    const fileNameNoExt = fileName.replace(/\.[^/.]+$/, ``)
+    const defaultAlt = fileNameNoExt.replace(/[^A-Z0-9]/gi, ` `)
+
+    // Construct new image node w/ aspect ratio placeholder
+    let rawHTML = `
+  <span
+    class="gatsby-resp-image-wrapper"
+    style="position: relative; display: block; ${
+      options.wrapperStyle
+    }; max-width: ${presentationWidth}px; margin-left: auto; margin-right: auto;"
+  >
+    <span
+      class="gatsby-resp-image-background-image"
+      style="padding-bottom: ${ratio}; position: relative; bottom: 0; left: 0; background-image: url('${
+      responsiveSizesResult.base64
+    }'); background-size: cover; display: block;"
+    >
+      <img
+        class="gatsby-resp-image-image"
+        style="width: 100%; height: 100%; margin: 0; vertical-align: middle; position: absolute; top: 0; left: 0; box-shadow: inset 0px 0px 0px 400px ${
+          options.backgroundColor
+        };"
+        alt="${node.alt ? node.alt : defaultAlt}"
+        title="${node.title ? node.title : ``}"
+        src="${fallbackSrc}"
+        srcset="${srcSet}"
+        sizes="${responsiveSizesResult.sizes}"
+      />
+    </span>
+  </span>
+  `
+    return rawHTML
+  }
+  return Promise.all(
+    // Simple because there is no nesting in markdown
+    markdownImageNodes.map(
+      node =>
+        new Promise(async (resolve, reject) => {
+          // Ignore gifs as we can't process them,
+          // svgs as they are already responsive by definition
```

Comment:
are gifs and svgs actually ignored here? if so explanation that `images.ctfassets.net` isn't used for those would be helpful